bindings/go/src/ucx/ucs_constants.go

```diff
@@ -34,6 +36,16 @@ func (m UcsMemoryType) String() string {
 	return C.GoString(C.ucxgo_get_ucs_mem_type_name(C.ucs_memory_type_t(m)))
 }
 
+func (m *UcsMemoryType) Set (value string) error {
+	cValue := unsafe.Pointer(C.CString(value))
+	res := C.ucxgo_parse_ucs_mem_type_name(cValue)
+	if res == -1 {
+		return errors.New("memory type can be either host or cuda")
+	}
+	*m = UcsMemoryType(res)
+	return nil
+}
```

Comment:
does it mean we do string search for each send operation?

Reply:
no, when converting from string which we do when parsing command-line args